homeassistant/components/mqtt/device_tracker/schema_yaml.py

```diff
@@ -26,13 +34,28 @@
 )
 
 
-async def async_setup_scanner_from_yaml(hass, config, async_see, discovery_info=None):
+async def async_setup_scanner_from_yaml(
+    hass: HomeAssistant, config, async_see, discovery_info=None
+):
     """Set up the MQTT tracker."""
     devices = config[CONF_DEVICES]
     qos = config[CONF_QOS]
     payload_home = config[CONF_PAYLOAD_HOME]
     payload_not_home = config[CONF_PAYLOAD_NOT_HOME]
     source_type = config.get(CONF_SOURCE_TYPE)
+    config_entry = hass.config_entries.async_entries(mqtt.DOMAIN)[0]
+    subscriptions: list[Callable] = []
+
+    @callback
+    def _entry_unload(*_: Any) -> None:
+        """Handle the unload of the config entry."""
+        # Unsubscribe from mqtt
+        for unsubscribe in subscriptions:
+            unsubscribe()
+        # Cleanup entity state
+        for dev_id in devices.keys():
+            entity_id = f"{DOMAIN}.{dev_id}"
+            hass.states.async_remove(entity_id)
```

Comment:
Why do we want to do this? There could be other (non-MQTT) device trackers tied to the same devices?